research/2d-decoding/rectangle_detector.py

```diff
@@ -4,7 +4,8 @@
 import matplotlib.pyplot as plt
 import cv2
 
-LINE_PIXEL_TOLERANCE = 1
+LINE_PIXEL_TOLERANCE = 5
```

Comment:
why 5?

Reply:
Because 1 was too aggressive and didn't accept rectangles in pictures, and 10 let in way too much. A dynamic approach based on the size of the region isn't applicable because this tolerance accounts for minor math/rounding mistakes in the line functions.